File: problems 1 to 10/p6 union and intersection/sol.java
class sol{

    static void swap(int arr[], int a, int b){
        int temp=arr[a];
        arr[a]=arr[b];
        arr[b]=temp;
    }
    static int partition(int arr[], int head, int tail){
        int index=head;
        int pivot=arr[head];
        for(int i=head+1;i<=tail;i++){
            if(pivot>=arr[i]){
                index++;
                swap(arr, index, i);
            }
        }
        swap(arr,index,head);
        return index;
    }
    static void Qsort(int arr[],int head,int tail){
        if(head<tail+1){
            int index=partition(arr, head, tail);
            Qsort(arr,head,index-1);
            Qsort(arr,index+1,tail);
    }
    }
    static void printarr(int arr[]){
        int size=arr.length;
        for(int i=0;i<size;i++){
                if(i==0) System.out.print("[");
                System.out.print(arr[i]);
                if(i!=size-1) System.out.print(" , ");
                if(i==size-1) System.out.print("]\n");
            }
        
}
    static int onion(int arr[],int add[]){
        int adday[]=new int[arr.length+add.length];

        for(int i=0;i<arr.length;i++) adday[i]=arr[i];
        for(int i=0;i<add.length;i++) adday[i+arr.length]=add[i];
    
        Qsort(adday,0,adday.length-1);     
        int count=1;
        for(int i=1;i<adday.length;i++){
            if(adday[i]!=adday[i-1]) {
                count++;
            }
        }
        return count;
    }
    public static void main(String[] args) {
        int arr[]={1,3,100,4,5,6};
        int add[]={0,8,2,2,1,2};
        int ans=onion(arr, add);
        System.out.print( ans);
    }
}

File: problems 1 to 10/p9 lowest distance in tower/sol.cpp
#include<iostream>
using namespace std;
class Solution {
  public:
    void swap(int *a,int *b){
    int temp=*a;
    *a=*b;
    *b=temp;
    }
    int partition(int *arr,int head,int tail){
        int index=head;
        int pivot=arr[head];
        for (int i = head+1; i <=tail; i++)
        {
            if(arr[i]<pivot){
                index++;
                swap(&arr[i],&arr[index]);
            }
        }
        swap(&arr[index],&arr[head]);
        return index;
    }
        
    void Qsort(int *arr, int head,int tail){
        if(head<tail){
            int index=partition(arr,head,tail);
            Qsort(arr,head,index-1);
            Qsort(arr,index+1,tail);
        }
    }
    
    int getMinDiff(int *arr, int n, int k){
    Qsort(arr,0,n);
    
    for(int i=0;i<n;i++){
        if(i<=(n-1)/2)
        {
            arr[i]=arr[i]+k;
        }else{
            arr[i]=arr[i]-k;
        }
    }
    
    return arr[n-1]-arr[0];
    }
};

int main(){
    int arr[]={7, 7 ,3, 5};
    Solution sol;
    cout<<sol.getMinDiff(arr,4,1);
    return 0;
}